### ISSUE TITLE
Unhandled type error when data field exists and data value is null

### ISSUE THREAD

**Issue by haxorof** [chop-dbhi/prometheus-sql#10]
It seems that if a SQL statement executes successfully and return with null the following error is seen in the logs:
```
Error setting metrics: Unhandled type %!s(<nil>)
```
This error is not that which is also easily fixed by doing a nil check on `dataVal` in [set.go](https://github.com/chop-dbhi/prometheus-sql/blob/master/set.go#L97).

**Comment by wangyanbn** [comment]
we can fix the SQL statement, make it returning the proper value instead of null. 
e.g.  `select NVL(FieldMayBeNull,0) as FieldNotNull from SomeTable`

**Comment by haxorof** [comment]
This is the query I ran when I saw this error in the log:
```yaml
- transaction_avg:
    driver: mysql
    connection:
        host: mysql
        port: 3306
        user: root
        password: unsecure
        database: test
    sql: >
         select avg(trans_end - trans_start) AS trans_avg from business_trans
            where trans_end is not null and trans_end between adddate(CURRENT_TIMESTAMP, INTERVAL -1 MINUTE) and CURRENT_TIMESTAMP;
    interval: 5s
    data-field: trans_avg
```
I will have to check if NVL is possible to use when not rows are returned.

Anyway, in the scenario when data field is available and data value is null the error log is not that clear that the query just returned null which is not supported to handle at the time. Introducing a data value  check for nil and just print it out that data was null will be a very small change but more understandable for the user of prometheus-sql what to fix. :)

**Comment by pschou** [comment]
This may be addressed in #66 with mapping nil to NaN.
